src/time-picker/time-picker.jsx

```diff
@@ -44,23 +44,33 @@ let TimePicker = React.createClass({
 
   formatTime(date) {
     let hours = date.getHours();
-    let mins = date.getMinutes();
-    let aditional = "";
+    let mins = date.getMinutes().toString();
 
     if (this.props.format === "ampm"){
       let isAM = hours < 12;
       hours = hours % 12;
-      aditional += isAM ? " am" : " pm";
-      hours = hours || 12;
+      let additional = isAM ? " am" : " pm";
+      hours = (hours || 12).toString();
+
+      if (mins.length < 2 ) mins = "0" + mins;
+
+      // Treat midday/midnight specially http://www.nist.gov/pml/div688/times.cfm
+      if (hours === "12" && mins === "00" && additional === " am") {
+        return "12 midday";
```

Comment:
Should this be 12 noon? Also, I think it would be a good idea to add a prop to switch this feature on and off.